posts/zzz_DO_NOT_EDIT_hello_admiralmetabolic/hello_admiralmetabolic!.qmd

```diff
@@ -0,0 +1,145 @@
+
+# Package Contents
+
+For this first 0.1 release, {admiralmetabolic} focuses primarily on the obesity indication. The package offers:
+
+* SDTM obesity-specific test data (to be moved to {pharmaversesdtm} at the next release of the package);
+* Two brand-new functions: admiralmetabolic::derive_param_waisthgt() and admiralmetabolic::derive_param_waisthip();
+* Two vignettes for an [obesity vital signs ADaM ADVS](https://pharmaverse.github.io/admiralmetabolic/articles/advs.html) and a [Control of Eating Questionnaire ADaM ADCOEQ](https://pharmaverse.github.io/admiralmetabolic/articles/adcoeq.html);
+* Two template programs for [ADVS](https://github.com/pharmaverse/admiralmetabolic/blob/main/inst/templates/ad_advs.R) and [ADCOEQ](https://github.com/pharmaverse/admiralmetabolic/blob/main/inst/templates/ad_adcoeq.R) matching the corresponding vignettes.
+
+## New Functions
+
+The first function, admiralmetabolic::derive_param_waisthgt(), is a wrapper to admiral::derive_param_computed() and enables users to add derived waist to height ratio parameters to a vital signs ADVS ADaM using waist circumference and height records. Here is an example of the function in action:
+
+```{r, eval = TRUE}
+library(tibble)
+library(admiralmetabolic)
+
+# Example 1: Derive Waist to Height Ratio where Height is measured only once
+
+advs <- tribble(
+  ~USUBJID, ~PARAMCD, ~PARAM, ~AVAL, ~AVALU, ~VISIT,
+  "01-101-1001", "HEIGHT", "Height (cm)", 147, "cm", "SCREENING",
+  "01-101-1001", "WSTCIR", "Waist Circumference (cm)", 110, "cm", "SCREENING",
+  "01-101-1001", "WSTCIR", "Waist Circumference (cm)", 108, "cm", "WEEK 2",
+  "01-101-1001", "WSTCIR", "Waist Circumference (cm)", 107, "cm", "WEEK 3",
+  "01-101-1002", "HEIGHT", "Height (cm)", 163, "cm", "SCREENING",
+  "01-101-1002", "WSTCIR", "Waist Circumference (cm)", 120, "cm", "SCREENING",
+  "01-101-1002", "WSTCIR", "Waist Circumference (cm)", 118, "cm", "WEEK 2",
+  "01-101-1002", "WSTCIR", "Waist Circumference (cm)", 117, "cm", "WEEK 3",
+)
+
+derive_param_waisthgt(
+  advs,
+  by_vars = exprs(USUBJID, VISIT),
+  wstcir_code = "WSTCIR",
+  height_code = "HEIGHT",
+  set_values_to = exprs(
+    PARAMCD = "WAISTHGT",
+    PARAM = "Waist to Height Ratio"
+  ),
+  constant_by_vars = exprs(USUBJID),
+  get_unit_expr = admiral::extract_unit(PARAM)
+)
+```
+
+Note that this function has built-in support to:
+
+* Calculate the ratio in cases where units differ across the numerator and denominator by converting on the fly (allowed units: m, mm, ft, in);
+* Work in cases such as pediatric studies where both waist circumference and height are measured at multiple visits, by modifying the argument `constant_by_vars`. Please visit the function's reference page for more details.
```

Comment:
```suggestion
Note that this function has built-in support for:

* Unit conversion in cases where units differ across the numerator and denominator (allowed units: m, mm, ft, in);
* Single or multiple height measurement by modifying the argument `constant_by_vars`. 

Please visit the function's reference page for more details.
```

For the constant_by_var, the default is actually that height is measured at every visit. Maybe this could be changed in the future, but I think we need to refer to it a different way to highlight this.

Reply:
updated